code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,360 @@
+/////Medi-Gun, coded by Deity Link, inspired by Valve's Team Fortress 2/////
+/obj/item/medigunpack
+	name = "\improper Medi-Gun"
+	desc = "You wear this on your back and heal people with it."
+	icon = 'icons/obj/gun_experimental.dmi'
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/backpacks_n_bags.dmi', "right_hand" = 'icons/mob/in-hand/right/backpacks_n_bags.dmi')
+	icon_state = "medigunfull"
+	item_state = "medigun"
+	w_class = W_CLASS_LARGE
+	flags = FPRINT
+	slot_flags = SLOT_BACK
+	var/obj/item/medigun/medigun = null
+	var/emagged = 0//emagged gun slowly deals damage of all types at once.
+
+/obj/item/medigunpack/examine(mob/user)
+	..()
+	if(medigun)
+		to_chat(user,"<span class=\"info\">Charge = [medigun.ubercharge]%</span>")
+
+/obj/item/medigunpack/New()
+	..()
+	medigun = new(src)
+
+/obj/item/medigunpack/Destroy()
+	if(medigun)
+		qdel(medigun)
+	medigun = null
+	..()
+
+/obj/item/medigunpack/MouseDrop(atom/over_object)
+	if(ishuman(usr))
+		var/mob/living/carbon/human/H = usr
+		if (!( istype(over_object, /obj/screen/inventory) ))
```

Comment:
That's what I get for copy pasting stuff at 4am with nothing in the stomach and a splitting headache.